apps/meteor/server/startup/migrations/v320.ts

```diff
@@ -0,0 +1,108 @@
+import type { IRole, IUser } from '@rocket.chat/core-typings';
+import { ROOM_ROLE_PRIORITY_MAP } from '@rocket.chat/core-typings';
+import { Users, Subscriptions } from '@rocket.chat/models';
+
+import { addMigration } from '../../lib/migrations';
+
+export const getRoomRolePriorityForRole = (role: IRole['_id']) =>
+	(ROOM_ROLE_PRIORITY_MAP as { [key: IRole['_id']]: number })[role] ?? ROOM_ROLE_PRIORITY_MAP.default;
+
+export const calculateRoomRolePriorityFromRoles = (roles: IRole['_id'][]) => {
+	return roles.reduce((acc, roleId) => Math.min(acc, getRoomRolePriorityForRole(roleId)), ROOM_ROLE_PRIORITY_MAP.default);
+};
+
+const READ_BATCH_SIZE = 1000;
+const WRITE_BATCH_SIZE = 1000;
+
+async function assignRoomRolePrioritiesFromMap(userIdAndroomRolePrioritiesMap: Map<IUser['_id'], IUser['roomRolePriorities']>) {
```

Comment:
Can you take a look at https://www.mongodb.com/docs/manual/reference/method/db.collection.initializeUnorderedBulkOp/#behavior and see if it works better with less code? Mongo says it kinda groups the operations https://www.mongodb.com/docs/v6.0/reference/method/Bulk/